test/mysql_tests.erl

```diff
@@ -279,6 +280,53 @@ basic_queries(Pid) ->
 
     ok.
 
+filtermap_queries(Pid) ->
+    ok = mysql:query(Pid, ?create_table_t),
+    ok = mysql:query(Pid, <<"INSERT INTO t (id, tx) VALUES (1, 'text 1')">>),
+    ok = mysql:query(Pid, <<"INSERT INTO t (id, tx) VALUES (2, 'text 2')">>),
+    ok = mysql:query(Pid, <<"INSERT INTO t (id, tx) VALUES (3, 'text 3')">>),
+
+    Query = <<"SELECT id, tx FROM t">>,
+
+    %% one-ary filtermap fun
+    FilterMap1 = fun
+        ([1|_]) ->
+            true;
+        ([2|_]) ->
+            false;
+        (Row1=[3|_]) ->
+            {true, list_to_tuple(Row1)}
+    end,
+
+    %% two-ary filtermap fun
+    FilterMap2 = fun
+        (_, Row2) ->
+            FilterMap1(Row2)
+    end,
+
+    Expected = [{3, <<"text 3">>}, [1, <<"text 1">>]],
+
+    %% test with plain query
+    {ok, _, Rows1}=mysql:query(Pid, Query, FilterMap1),
+    ?assertEqual(Expected, lists:sort(Rows1)),
```

Comment:
I suggest you add `ORDER BY id` to the SQL query instead of these calls to `lists:sort/1`. Relying on the fact that tuples < lists in erlang isn't the most obvious thing.

Reply:
Ok, no problem :)